akka-remote/src/test/scala/akka/remote/artery/FlushOnShutdownSpec.scala

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import akka.actor.{ Actor, ActorIdentity, ActorSystem, Identify, Props, RootActorPath }
+import akka.remote.RARP
+import akka.testkit.{ AkkaSpec, ImplicitSender, TestProbe }
+import com.typesafe.config.ConfigFactory
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+object FlushOnShutdownSpec {
+
+  val config = ConfigFactory.parseString(s"""
+     akka {
+       actor.provider = remote
+       actor.serialize-creators = off
+       remote.artery.enabled = on
+       remote.artery.canonical.hostname = localhost
+       remote.artery.canonical.port = 0
+     }
+  """)
+
+}
+
+class FlushOnShutdownSpec extends AkkaSpec(FlushOnShutdownSpec.config) {
```

Comment:
`ArteryMultiNodeSpec` is very nice for this kind of test with more than one ActorSystem